src/middlewared/middlewared/plugins/pool.py

```diff
@@ -333,7 +339,54 @@ class Config:
 
     @filterable
     async def query(self, filters, options):
-        return await self.middleware.call('zfs.dataset.query', filters, options)
+        return await self.middleware.run_in_thread(
```

Comment:
Why `run_in_thread`? I don't see any blocking IO / heavy CPU stuff in there.

Reply:
@themylogin having no IO means there will be no context switch for the event loop.

It might not be heavy CPU but imagine a scenario with 1000+ datasets, if all of that run in a single coroutine it may block the event loop for a few hundred of milliseconds.
If we make `__transform` and `transform` coroutines we will have 1000+ context switches which might degrade performance.

Do you think thats a wrong assumption?

Reply:
@william-gr I agree with you